charts/argo-workflows/README.md

```diff
@@ -2,23 +2,79 @@
 
 ## Pre-Requisites
 
 ### Custom resource definitions
 
+#### Full vs Minified CRDs
+
+By default, this chart installs the **minified CRDs** (which use `x-kubernetes-preserve-unknown-fields` to accept any fields).
+These are smaller in size but provide almost no validation.
+This is the default in upstream manifests.
+
+If you prefer to use the **full CRDs** with complete OpenAPI schemas (recommended for better validation and type safety), you can set:
+
+```bash
+--set crds.full=true
+```
+
+**Important:** Full CRDs are approximately 7.4MB total and will most likely need to be applied using Server Side Apply to avoid hitting size limits.
+When installing manually with kubectl, use:
+
+```bash
+kubectl apply --server-side --force-conflicts -k "https://github.com/argoproj/argo-workflows/manifests/base/crds/full?ref=v<argoVersion>"
+```
+
+Note: Helm 3 does not natively support Server Side Apply.
+If you encounter issues installing full CRDs via Helm (such as "Too long" errors), you have two options:
+
+**Option 1: Manual kubectl apply**
+
+1. Apply the CRDs manually using kubectl with `--server-side` as shown above
+2. Install the chart with `--set crds.install=false` to skip CRD installation
+
+**Option 2: Use ArgoCD with Server Side Apply**
+
+ArgoCD supports Server Side Apply for Helm charts. To enable it, set the sync option in your Application manifest:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: argo-workflows
+spec:
+  syncPolicy:
+    syncOptions:
+    - ServerSideApply=true
+  source:
+    chart: argo-workflows
+    repoURL: https://argoproj.github.io/argo-helm
+    targetRevision: 0.xx.xx
+    helm:
+      values: |
+        crds:
+          full: true
+```
+
+#### Installing CRDs Outside the Chart
+
 Some users would prefer to install the CRDs _outside_ of the chart. You can disable the CRD installation of this chart by using `--set crds.install=false` when installing the chart.
 
 Helm cannot upgrade custom resource definitions in the `<chart>/crds` folder [by design](https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations). Starting with 3.4.0 (chart version 0.19.0), the CRDs have been moved to `<chart>/templates` to address this design decision.
 
-If you are using Argo Workflows chart version prior to 3.4.0 (chart version 0.19.0) or have elected to manage the Argo Workflows CRDs outside of the chart, please use `kubectl` to upgrade CRDs manually from [templates/crds](templates/crds/) folder or via the manifests from the upstream project repo:
+If you are using Argo Workflows chart version prior to 3.4.0 (chart version 0.19.0) or have elected to manage the Argo Workflows CRDs outside of the chart, please use `kubectl` to upgrade CRDs manually from [templates/crds-full](templates/crds-full/) or [templates/crds-minified](templates/crds-minified/) folders, or via the manifests from the upstream project repo:
```

Comment:
I believe it should point to `files/crds/full` and `files/crds/minimal`. Otherwise people may get confuse and try to apply those templates directly.

Reply:
Yes, good catch. Done